File: Test-Smells-Game/Projects/Leaderboard_service/src/main/java/com/dariotintore/tesi/leaderboard/entity/Comment.java
package com.dariotintore.tesi.leaderboard.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "comment")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Comment {
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO, generator="seq")
  @GenericGenerator(name = "seq", strategy="increment")
  @Column(name = "comment_id")
  private Long commentId;

  @Column(name = "author")
  private String commentAuthor;

  @Column(name = "text")
  private String commentText;
  @Column(name = "solution_id")
  private Long solutionId;

  @Column(name = "creation_timestamp")
  @CreationTimestamp
  private Date creationTimestamp;
}

File: Test-Smells-Game/Projects/Frontend_electron/src/app/model/comment/comment.ts
export class SolutionComment {
  commentId!: number;
  commentAuthor!: string;
  commentText!: string;
  solutionId!: number;
  creationTimestamp!: Date;

  constructor(commentAuthor: string, commentText: string) {
    this.commentAuthor = commentAuthor;
    this.commentText = commentText;
  }
}
